### PR TITLE
Szeged index

### PR DESCRIPTION
**Pull request by jangwon-yie**
It contains an implementation of Szeged index(https://en.wikipedia.org/wiki/Szeged_index) and its unit tests.

### CODE REVIEW

File: networkx/algorithms/szeged.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    -----
+    For an edge e = ij, let n_e(i) be the number of vertices of G being closer
+    to i than to j and let n_e(j) be the number of vertices of G lying closer
+    to j than to i. The Szeged index of a graph G is defined by that the index
+    is sum of the product of n_e(x) and n_e(y) for any e = xy in E(G). Formally,
+    szeged(G) = sum([n_e(x) * n_e(y) for e=xy in E(G)])
+
+    More information can be found at:
+    https://en.wikipedia.org/wiki/Szeged_index
+
```
Comment (MridulS) [inline networkx/algorithms/szeged.py:44]: A references section would be nice:

```
References
------------

.. [1] Gutman, Ivan. "A formula for the Wiener number of trees and its extension to graphs containing cycles." Graph Theory Notes NY 27, no. 9 (1994): 9-15.
```

File: networkx/algorithms/szeged.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    v_dist = {w: dist for w, dist in spl(g, source=v).items()}
+
+    vertices = list(g.nodes)
+    vertices.remove(u)
+    vertices.remove(v)
+
+    u_index = len([w for w in vertices if u_dist[w] < v_dist[w]])
+    v_index = len([w for w in vertices if v_dist[w] < u_dist[w]])
+
+    return u_index * v_index
```
Comment (MridulS) [inline networkx/algorithms/szeged.py:74]:
```suggestion
    u_dist = spl(g, source=edge[0])
    v_dist = spl(g, source=edge[1])

    vertices = g.nodes - edge

    u_index = len([w for w in vertices if u_dist[w] < v_dist[w]])
    v_index = len([w for w in vertices if v_dist[w] < u_dist[w]])

    return u_index * v_index
```

Maybe something like this?

**Review by MridulS — COMMENTED** [review]
@jangwon-yie it has been a looooong time and sorry for taking so much time in getting this reviewed. I would love to get this in!

Please do let me know if you have the bandwidth to work on this through the review, no worries if not. I can cherry pick the commits and start a new PR :)

**Comment by MridulS** [comment]
@jangwon-yie let me know if you have the bandwidth to finish this up :) Otherwise I'll go ahead and close this PR. Thanks for your contributions!

**Comment by rossbar** [comment]
I'm going to go ahead and close this. If someone wants to pick it up again please feel free.

**Pull request closed without merging**